### DISCUSSION TITLE
Question about JetStream streams

### DISCUSSION THREAD

**Discussion by boskiv** [nats-io/nats.py#851] [Q&A]
So there is no way to subscribe to the stream from client? 
I need to create a consumer and poll it in loop for the messages?
And there is no history in pub/sub model? 
What should I do if I need both ?

On connect I need to replay 10 days messages for example and subscribe for a new messages coming in realtime. 
What is my options ?

**Comment by caspervonb** [comment]
You can use `js.subscribe()` which creates a push-based consumer under the hood — no manual polling needed. To replay historical messages and then continue receiving new ones in real-time, use `deliver_policy=DeliverPolicy.ALL` (or a specific start time with `DeliverPolicy.BY_START_TIME` and `opt_start_time`):

```python
nc = await nats.connect()
js = nc.jetstream()

# Subscribe with full history replay, then continue in real-time
sub = await js.subscribe("my.subject", durable="my-consumer")

async for msg in sub.messages:
    print(msg.data)
    await msg.ack()
```

For a 10-day replay window specifically:

```python
from nats.js.api import ConsumerConfig, DeliverPolicy
from datetime import datetime, timedelta, timezone

start = datetime.now(timezone.utc) - timedelta(days=10)
sub = await js.subscribe(
    "my.subject",
    durable="my-consumer",
    config=ConsumerConfig(deliver_policy=DeliverPolicy.BY_START_TIME, opt_start_time=start),
)
```

Moving this to Discussions as it's a usage question.
